File: 5.prefix_sums/lesson_5_passing_cars.py
def solution(A):
    n = len(A)
    if n == 1:
        return 0

    #reverse list
    A = A[::-1]
    
    prefix_sums = [0] * n
    prefix_sums[0] = A[0]

    for i in range(1, n):
        prefix_sums[i] = prefix_sums[i-1] + A[i]

    count = 0
    for index, elem in enumerate(A):
        if(elem == 0):
            count += prefix_sums[index]
        
    if(count <= 1000000000):
        return count

    return -1

File: 5.prefix_sums/lesson_5_passing_cars.js
function solution(A) {
    let n = A.length
    let pairs = 0
    let prefSums = new Array(n)
    prefSums[0] = A[0]
    
    for(let i = 1; i < n; i++) {
        prefSums[i] = prefSums[i-1] + A[i]
    }

    for(let i = 0; i < n; i++) {
        if(A[i] == 0) {
            pairs += (prefSums[n-1] - prefSums[i])
        }
    }

    if(pairs > 1000000000) {
        return -1
    }
    
    return pairs
}